packages/p2p/src/components/composite-calendar/composite-calendar-mobile.jsx

```diff
@@ -0,0 +1,253 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Button, DatePicker, Icon, InputField, MobileDialog, Text } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { toMoment } from '@deriv/shared';
+import './composite-calendar-mobile.scss';
+
+export const RadioButton = ({ id, className, selected_value, value, label, onChange }) => {
+    return (
+        <label
+            htmlFor={id}
+            className={classNames('composite-calendar-modal__radio', className, {
+                'composite-calendar-modal__radio--selected': selected_value === value,
+            })}
+            onClick={() => onChange({ label, value })}
+        >
+            <input className='composite-calendar-modal__radio-input' id={id} type='radio' value={value} />
+            <span
+                className={classNames('composite-calendar-modal__radio-circle', {
+                    'composite-calendar-modal__radio-circle--selected': selected_value === value,
+                })}
+            />
+            <Text
+                as='p'
+                color='prominent'
+                size='xs'
+                line_height='unset'
+                weight={selected_value === value ? 'bold' : 'normal'}
+            >
+                {label}
+            </Text>
+        </label>
+    );
+};
+const CUSTOM_KEY = 'custom';
+
+const CompositeCalendarMobile = React.memo(
+    ({ input_date_range, current_focus, duration_list, onChange, setCurrentFocus }) => {
+        const date_range = input_date_range || duration_list.find(range => range.value === 'all_time');
+
+        const [from, setFrom] = React.useState(from && toMoment(from).format('DD MMM YYYY'));
```

Comment:
wait, what is this `from && toMoment(from).format('DD MMM YYYY')` does? isn't it will be null at first